### PR TITLE
Add EC2 IAM role gotcha

### PR DESCRIPTION
**Pull request by luqasz**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -739,6 +739,7 @@ EC2
 -	🔸Periodically you may find that your server or load balancer is receiving traffic for (presumably) a previous EC2 server that was running at the same IP address that you are handed out now (this may not matter, or it can be fixed by migrating to another new instance).
 -	❗If the EC2 API itself is a critical dependency of your infrastructure (e.g. for automated server replacement, custom scaling algorithms, etc.) and you are running at a large scale or making many EC2 API calls, make sure that you understand when they might fail (calls to it are [rate limited](http://docs.aws.amazon.com/AWSEC2/latest/APIReference/query-api-troubleshooting.html#api-request-rate) and the limits are not published and subject to change) and code and test against that possibility.
 -	❗Many newer EC2 instance types are EBS-only. Make sure to factor in EBS performance and costs when planning to use them.
+-	❗Once EC2 instance is launched, IAM role can not be assigned to it.
```
Comment (jlevy) [inline README.md]: Good note. Needs a link. Also I think it'd be clearer to phrase it
"🔸An IAM role can be assigned to an EC2 instance [only at launch time](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html). You cannot assign to an existing instance."

Or similar. Is there a better link?

Also please use 🔸 as the ❗is for really costly gotchas only.

**Comment by jlevy** [comment]
I'm good w these changes; adding @max-zanko for review.

File: README.md (hunk 1 above)
Comment (ThanosBaskous) [inline README.md]: 👍 to this feedback. I'd say "You cannot assign it to.." (inclusion of "it")

Comment (luqasz) [inline README.md]: I've found a different [link](http://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_switch-role-ec2.html)

File: README.md [hunk 2]
```diff
@@ -739,7 +739,7 @@ EC2
 -	🔸Periodically you may find that your server or load balancer is receiving traffic for (presumably) a previous EC2 server that was running at the same IP address that you are handed out now (this may not matter, or it can be fixed by migrating to another new instance).
 -	❗If the EC2 API itself is a critical dependency of your infrastructure (e.g. for automated server replacement, custom scaling algorithms, etc.) and you are running at a large scale or making many EC2 API calls, make sure that you understand when they might fail (calls to it are [rate limited](http://docs.aws.amazon.com/AWSEC2/latest/APIReference/query-api-troubleshooting.html#api-request-rate) and the limits are not published and subject to change) and code and test against that possibility.
 -	❗Many newer EC2 instance types are EBS-only. Make sure to factor in EBS performance and costs when planning to use them.
-
+-	🔸An IAM role can be assigned to an EC2 instance [only at launch time](http://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_switch-role-ec2.html). You cannot assign to an existing instance.
 AMIs
```
Comment (ThanosBaskous) [inline README.md]: I think you need a space before this new section header.

**Comment by ThanosBaskous** [comment]
lgtm, pending topic review by @max-zanko 

Thank you @luqasz!

**Comment by jlevy** [comment]
This is confused enough with the bad merge I think fixing it is too hard. Have re-done it in #242.

**Comment by jlevy** [comment]
Thanks @luqasz ! Your commits are merged in #242.

**Pull request closed without merging**